pkg/provisioner/ironic/ironic.go

```diff
@@ -1336,6 +1336,25 @@ func (p *ironicProvisioner) Delete() (result provisioner.Result, err error) {
 		return p.setMaintenanceFlag(ironicNode, true)
 	}
 
+	if p.host.Status.PoweredOn {
+		p.log.Info("host is being powered off before removal")
+		provResult, err := p.hardPowerOff()
+		switch {
+		case err == nil:
+			p.host.Status.PoweredOn = false
+			// Save power status change by marking result as Dirty
+			result.Dirty = true
+			return result, nil
+		case provResult.Dirty:
```

Comment:
I don't think we'll ever get here, because if there is an error (failing the first case) then Dirty is unlikely to be true.

Reply:
https://github.com/metal3-io/baremetal-operator/blob/master/pkg/provisioner/ironic/ironic.go#L1471 seems to be one place where err == nil and Dirty == true.

Reply:
Right, but if there is an error then err won't be nil.

Reply:
This code has been restructured.